mmv1/products/filestore/terraform.yaml

```diff
@@ -40,6 +40,8 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         ignore_read: true
       networks.reservedIpRange: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
+      networks.connectMode: !ruby/object:Overrides::Terraform::PropertyOverride
+        custom_flatten: 'templates/terraform/custom_flatten/default_if_empty.erb'
```

Comment:
it's not liking this when trying to compile:
```
Error compiling file: /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/templates/terraform/custom_flatten/default_if
_empty.erb
Error compiling file: /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/templates/terraform/flatten_property_method.e
rb
Error compiling file: /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/templates/terraform/flatten_property_method.e
rb
Error compiling file: /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1//templates/terraform/resource.erb
bundler: failed to load command: compiler (compiler)
RuntimeError: Unsupported go literal
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/google/golang_utils.rb:37:in `go_literal'
  (erb):4:in `compile_template'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `eval'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `result'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:219:in `compile_string'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:154:in `compile_file'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/provider/terraform/sub_template.rb:93:in `compile_template'
  (erb):3:in `compile_template'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `eval'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `result'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:219:in `compile_string'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:154:in `compile_file'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/provider/terraform/sub_template.rb:93:in `compile_template'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/provider/terraform/sub_template.rb:37:in `build_flatten_method'
  (erb):123:in `block in compile_template'
  (erb):122:in `each'
  (erb):122:in `compile_template'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `eval'
  /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/erb.rb:901:in `result'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:219:in `compile_string'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/compile/core.rb:154:in `compile_file'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/provider/terraform/sub_template.rb:93:in `compile_template'
  /Users/stoelinga/go/src/github.com/GoogleCloudPlatform/magic-modules/mmv1/provider/terraform/sub_template.rb:37:in `build_flatten_method'
  (erb):956:in `block in provider_binding'
  (erb):955:in `each'
  (erb):955:in `provider_binding'
```

Debugging this since the error message indicates the error is from my code

Reply:
I think I found the issue. default_if_empty will use the property default_value and I wasn't setting default_value anymore hence the error.